api/subscriptions/block_reader_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package subscriptions
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/genesis"
+	"github.com/vechain/thor/v2/muxdb"
+	"github.com/vechain/thor/v2/packer"
+	"github.com/vechain/thor/v2/state"
+	"github.com/vechain/thor/v2/thor"
+	"github.com/vechain/thor/v2/tx"
+	"github.com/vechain/thor/v2/txpool"
+)
+
+func TestBlockReader_Read(t *testing.T) {
+	repo, generatedBlocks, _ := initChain(t)
+	genesisBlk := generatedBlocks[0]
+	newBlock := generatedBlocks[1]
+
+	// Test case 1: Successful read next blocks
+	br := newBlockReader(repo, genesisBlk.Header().ID())
+	res, ok, err := br.Read()
+
+	assert.NoError(t, err)
+	assert.True(t, ok)
+	if resBlock, ok := res[0].(*BlockMessage); !ok {
+		t.Fatal("unexpected type")
+	} else {
+		assert.Equal(t, newBlock.Header().Number(), resBlock.Number)
+		assert.Equal(t, newBlock.Header().ParentID(), resBlock.ParentID)
+	}
+
+	// Test case 2: There is no new block
+	br = newBlockReader(repo, newBlock.Header().ID())
+	res, ok, err = br.Read()
+
+	assert.NoError(t, err)
+	assert.False(t, ok)
+	assert.Empty(t, res)
+
+	// Test case 3: Error when reading blocks
+	br = newBlockReader(repo, thor.MustParseBytes32("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"))
+	res, ok, err = br.Read()
+
+	assert.Error(t, err)
+	assert.False(t, ok)
+	assert.Empty(t, res)
+}
+
+func initChain(t *testing.T) (*chain.Repository, []*block.Block, *txpool.TxPool) {
+	db := muxdb.NewMem()
+	stater := state.NewStater(db)
+	gene := genesis.NewDevnet()
+
+	b, _, _, err := gene.Build(stater)
+	if err != nil {
+		t.Fatal(err)
+	}
+	repo, _ := chain.NewRepository(db, b)
+
+	txPool := txpool.New(repo, stater, txpool.Options{
```

Comment:
Add this to discard logger output https://github.com/vechain/thor/blob/d903f1f312368d2cf843c5e7a132246cb6093075/api/transactions/transactions_test.go#L37-L39

Reply:
For the same package, we only need it once.